internal/app/config_preference.go

```diff
@@ -81,6 +84,56 @@ func (a *app) loadPreferencesConfig() {
 	a.ConfigRegister(cfg)
 
 	window.SetFps(cfg.Prefs.Interface.Fps)
+
+	// Ensure code editor is possible otherwise default
+	configFilePath := configFilePath(a.configDir, cfg.Name())
+	switch cfg.Prefs.Editor.CodeEditor {
+	case prefs.CodeEditorVSC:
+		if _, err := exec.LookPath(prefs.CodeEditorVSCActual); err != nil {
+			if _, err := exec.LookPath(prefs.CodeEditorDMActual); err == nil {
+				// VSC invalid but DM valid
+				cfg.Editor.CodeEditor = prefs.CodeEditorDM
+			} else if _, err := exec.LookPath(prefs.CodeEditorNPPActual); err == nil {
+				// VSC invalid but NPP valid
+				cfg.Editor.CodeEditor = prefs.CodeEditorNPP
+			} else {
+				// All invalid just default
+				cfg.Editor.CodeEditor = prefs.CodeEditorDefault
+			}
+			log.Print(prefs.CodeEditorVSCActual + " could not be found in PATH. Code Editor pref changed to " + cfg.Editor.CodeEditor)
+			config.SaveV(configFilePath, cfg)
```

Comment:
You don’t need to save the configuration manually, as there’s already an automatic saving mechanism in place.